Localization support and transifex integration

NaitLee opened the pull request:
This contributes to progress of #54 by the following additions:
- Script `lang/makepot.sh` for gathering & updating strings from source code (*of all branches*) for translation, and it excludes vanilla strings from this codebase
- Script `lang/merge.sh` for “merging” translations: merge vanilla strings when packaging for distribution as a whole as a pk3 (will also be run by `build.sh` by default)
- Script `lang/updatepo.sh` for updating existing po files manually, useful for offline translation
- Transifex configuration `.transifex.yml` for integration with the service
- Some initial Chinese translations are also included

Before merge, check if the scripts work on your machine… :)

After merging, consider:
- Create *your own team* on transifex, set this project up with provided configuration, and put the project link to readme
- (or probably I can manage a team, if you want)
- Run `lang/makepot.sh`, to have new strings (those from other branches) included for translation; consider running it routinely
- Make more strings translatable; move server side strings to client if necessary; some short strings may be better to have prefix for precious meaning
- After some time, make an optional content pack with newest translations for the mod by running `lang/merge.sh`, and consider to distribute it

If I missed something just ask… :)

NaitLee commented:
@LegendaryGuard just have added you to the testing transifex team. I was busy for a very long time and didn't take time to manage that.

LegendaryGuard commented:
> @LegendaryGuard just have added you to the testing transifex team. I was busy for a very long time and didn't take time to manage that.

Nice! Right now, I just entered into the testing transifex project and I translated few japanese strings to see if it updates in the next time.
How long does that last to be updated? Every (2, 3, 4, ...) days, a week, a month, or...? I don't have clue how it's configured.

NaitLee commented:
> How long does that last to be updated? Every (2, 3, 4, ...) days, a week, a month, or...? I don't have clue how it's configured.

I just modified the config to 1% translated before pushing to repo. It's been 50%, though I noted that the updated pot contains so many strings... :)

I see a push was issued after I added a string afterwards, with some minutes delay. (I set a 5 mins cooldown as for now)

LegendaryGuard commented:
> > How long does that last to be updated? Every (2, 3, 4, ...) days, a week, a month, or...? I don't have clue how it's configured.
> 
> I just modified the config to 1% translated before pushing to repo. It's been 50%, though I noted that the updated pot contains so many strings... :)
> 
> I see a push was issued after I added a string afterwards, with some minutes delay. (I set a 5 mins cooldown as for now)

Ok. We think that the transifex bot should check once a week, what do you think, @MarioSMB ? Or should be once a month? If not, how long do you prefer for the bot to check the updates in transifex of some languages and make a commit?

P.D. Transifex bot commit from right now: Nice, it updates correctly as expected!

LegendaryGuard commented:
I took a look to some details, these words aren't from the [`main`](https://github.com/MarioSMB/modpack) branch:
- https://app.transifex.com/naitlee-soft/test-smbmod/translate/#ja_JP/0623e4e9ca81c758d238c3c848be8892/492711740?q=text%3ACyber
- https://app.transifex.com/naitlee-soft/test-smbmod/translate/#ja_JP/0623e4e9ca81c758d238c3c848be8892/492711747?q=text%3AZ-Saber
- https://app.transifex.com/naitlee-soft/test-smbmod/translate/#ja_JP/0623e4e9ca81c758d238c3c848be8892/492711726?q=text%3ADetective

Why do these words appear that come from another branch ([`cyber`](https://github.com/MarioSMB/modpack/tree/cyber))?

NaitLee commented:
> It would be ok to update the script in the PR.

done

> Why do these words from another branch ([cyber](https://github.com/MarioSMB/modpack/tree/cyber)) appear there?

`makepot.sh` gathers translatable strings from all mod branches, so that the translation can be used for all variation of mods.

LegendaryGuard commented:
Understood. You need to clarify that requires installed this package to execute the translations (`msgmerge` and `xgettext` commands):
- Linux:```sh
sudo apt-get update
sudo apt-get gettext
``` 
- Windows MSYS:```sh
pacman -S gettext
```

Ok, so the manual process is:
- Go to `lang/` directory
- Execute: `./makepot.sh`
- Execute: `./merge.sh`
- Copy/Move `zzz-smb-l10n*.pk3 `into `xonotic-data.pk3dir/data`, `Xonotic/data`
- Run the game and enjoy with the translated languages from the mod!
- Execute if there are new translations: `./updatepo.sh`

`updatepo.sh` should remove the `*.po~` files if needed, I think.

Btw, I created a new organization to attach the modpack: https://app.transifex.com/team-xonsmb/

Here the empty created project to attach: https://app.transifex.com/team-xonsmb/xonotic-smb-modpack/dashboard/

So, I think you can replace that line in the README to the URL I just shared: https://github.com/MarioSMB/modpack/pull/55/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R57

If you do, consider updating the documentation I just shared.

And I guess the instructions to execute successfully from the organization are there, right? https://github.com/MarioSMB/modpack/pull/55#issue-2125819503

NaitLee commented:
Things are updated.

`./updatepo.sh` are intended for those who translate via editing .po files directly (such as me)  
(and they just upload .po files to Transifex)  
Code maintainers don't have to run it, because updates to .po files are done by Transifex.

To avoid pull request spams, you may configure Transifex to directly push translations to master branch.

MarioSMB commented:
Thank you for continuing to work on this - I've given LegendGuardian permission to set it up at the earliest convenience.

LegendaryGuard commented:
@NaitLee Sorry for asking you some questions, but I want to make sure I apply everything correctly. Since the team and project have already been created, I'm not sure if I missed any steps in Transifex before running these scripts via GitHub Actions:

1. Is this ok?
<img src="https://github.com/user-attachments/assets/c21bbb2e-31b0-49f9-80af-067c93e9340f" width=600 alt="opensrcprojattach" />

2. Does GitHub integration need to be enabled? I think this step requires the owner's permission and the installation of the Transifex GitHub Actions.

<img src="https://github.com/user-attachments/assets/d00dbb14-cd38-4659-9f79-aa75bd7b7e02" width=600 alt="transifexghintegration" />

NaitLee commented:
@LegendaryGuard

The project url should be ok.

The "integration" should be installed, and should be confirmed by the GitHub account owning this repository, then probably configured with target .pot file(s):

![image](https://github.com/user-attachments/assets/d2b85921-8707-40ab-a82d-3f75153c4d3b)

Note that this integration isn't GitHub Action but GitHub "App" for a repository:

![image](https://github.com/user-attachments/assets/1703c5fc-1f52-4b96-9889-19acf8aaf09c)

(Nevertheless you may create a GitHub Action to automatically build translation pk3s. I'm not experienced to this yet.)
